AirCasting/MicrophoneSession/MicrophoneManager.swift

```diff
@@ -102,68 +94,36 @@ private extension MicrophoneManager {
         
         switch type {
         case .ended:
-            levelTimer = Timer.scheduledTimer(timeInterval: 1, target: self, selector: #selector(getMeasurement), userInfo: nil, repeats: true)
+            if !recorder.isRecording {
+                recorder.record()
+            }
+            levelTimer = Timer.scheduledTimer(timeInterval: 1, target: self, selector: #selector(timerTick), userInfo: nil, repeats: true)
         case .began:
             fallthrough
         @unknown default:
-            levelTimer.invalidate()
-            session!.status = .DISCONNETCED
+            levelTimer?.invalidate()
+            try! measurementStreamStorage.updateSessionStatus(.DISCONNETCED, for: session!.uuid)
```

Comment:
This was just moved from existing logic but I think this means that the interruption logic has stopped recording because of a call but this there is also connection code made after the call is finished.